src/model/implementation/object-type-base.ts

```diff
@@ -41,40 +66,59 @@ export abstract class ObjectTypeBase extends TypeBase {
         }
 
         this.validateDuplicateFields(context);
+        this.validateSystemFieldOverrides(context);
 
         for (const field of this.fields) {
             field.validate(context);
         }
     }
 
+    private validateSystemFieldOverrides(context: ValidationContext): void {
```

Comment:
I wonder if we can add some validation logic so all the regular field directives are not silently discarded. We could add a generic logic to `create-model.ts` so that for system fields, it only allows an allowlist of directives (so currently `@hidden` and `@key`) and reports an error if any of the others are given

We could also do it here, but then we would need to add a check for all features in the field input, and would need to think of adding it whenever a new feature with a new directive is added

Reply:
Currently the systemFields are declared twice, once for the `RootEntityType` and once for the `ChildEntityType`. Not quite sure though why they are separated, since they are identical.

So it is not really possible to check this in `create-model.ts` without knowing which system fields to refer to.

When we only have one source of truth for system fields, we could add the "allow list" to the `SystemFieldConfig` and perform the check for redundant overrides and not-allowed directives on the override in `create-model.ts`.

Reply:
yeah I got confused and somehow thought `createModel` would be called within `new RootEntityType()` etc. which does not make sense at all :D

> When we only have one source of truth for system fields

would still need to distinguish between root/child entities and entity extensions / value objects. And we might add a system field to root entities but not child entities later.

Another idea: just retrieve the directives from the `field.astNode` in `object-type-base.ts`? I think the implementation classes are easier to reason about than `create-model.ts `, so I would rather move the directive handling there than to move the system field logic into `createModel`.

Reply:
and similar reasoning as [above](https://github.com/AEB-labs/cruddl/pull/315#discussion_r1532268739) - making the validation errors dependent on the astNode is probably fine because they are not critical - there is no chance to produce an inconsistent model just by omitting the astNode, because we're just checking for directives that won't have an effect anyway

Reply:
Sounds good 👍 
I added the "allowList" to the `SystemFieldConfig`s and then test against them in the `object-type-base.ts`. This also makes the redundancy check easier since I can simply test for a system field override without any directive declared.